tests/constraints/models.py

```diff
@@ -18,6 +18,12 @@ class Meta:
                 check=models.Q(price__gt=0),
                 name='%(app_label)s_%(class)s_price_gt_0',
             ),
+            models.CheckConstraint(
+                check=models.expressions.RawSQL(
+                    '(price < %s)', (1000,), output_field=models.BooleanField()
```

Comment:
I'll have to figure out a way to generate valid Oracle SQL here.

Reply:
Boolean expressions are allowed in check constraints on Oracle, however currently we generate
```sql
CHECK (((price < 1000)) = 1)
```
which is invalid because you cannot compare to booleans, we should generate:
```sql
CHECK (price < 1000)
```
My first idea is to add conditional `RawSQL`'s to the `conditional_expression_supported_in_where_clause()` on Oracle, e.g.
```diff
diff --git a/django/db/backends/oracle/operations.py b/django/db/backends/oracle/operations.py
index 59f55dccd5..4811957177 100644
--- a/django/db/backends/oracle/operations.py
+++ b/django/db/backends/oracle/operations.py
@@ -5,7 +5,7 @@ from functools import lru_cache
 from django.conf import settings
 from django.db.backends.base.operations import BaseDatabaseOperations
 from django.db.backends.utils import strip_quotes, truncate_name
-from django.db.models.expressions import Exists, ExpressionWrapper
+from django.db.models.expressions import Exists, ExpressionWrapper, RawSQL
 from django.db.models.query_utils import Q
 from django.db.utils import DatabaseError
 from django.utils import timezone
@@ -636,4 +636,6 @@ END;
             return True
         if isinstance(expression, ExpressionWrapper) and isinstance(expression.expression, Q):
             return True
+        if isinstance(expression, RawSQL) and getattr(expression, 'conditional', False):
+            return True
         return False
```

It works for me, TBH it is not strictly related with this PR but rather with `#25367`. IMO we should consider this as a release blocker for Django 3.0. I'm going to add A ticket and prepare a patch.

Reply:
Ticket [30986](https://code.djangoproject.com/ticket/30986).

Reply:
and PR #12068.

Reply:
This could be switched to the following then

```suggestion
                    'price < %s', (1000,), output_field=models.BooleanField()
```

I added the parenthesis wrapping to try to make it work on Oracle but that ended by being vain 😅